### PR TITLE
Add missing notification context to onNotification

### PR DESCRIPTION
**Pull request by guilhermebruzzi**
Add missing url and deeplink props from pinpoint notification

*Issue #, if available:*

Fix https://github.com/aws/aws-amplify/issues/1097

*Description of changes:*

This PR allows you to send notifications from Amazon Pinpoint with a url or deeplink params and have them on React Native context, so you can decide how your app react to different params.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react-native/src/PushNotification/PushNotification.js [hunk 1]
```diff
@@ -184,7 +184,9 @@ export default class PushNotification {
                 treatment_id: dataPayload['pinpoint.campaign.treatment_id']
             };
             const pinpoint = {
-                campaign
+                campaign,
+                deeplink: dataPayload['pinpoint.deeplink'],
+                url: dataPayload['pinpoint.url'],
```
Comment (mlabieniec) [inline packages/aws-amplify-react-native/src/PushNotification/PushNotification.js]: Should we just surface the entire `dataPayload`?

Comment (guilhermebruzzi) [reply packages/aws-amplify-react-native/src/PushNotification/PushNotification.js]: @mlabieniec Like a generic logic to parse the composed keys and value like `dataPayload['pinpoint.url'] = value` to `dataPayload.pinpoint = {}; dataPayload.pinpoint.url = value`?

It's a nice way to solve this, would you like me to change this PR to do that?

Comment (powerful23) [reply packages/aws-amplify-react-native/src/PushNotification/PushNotification.js]: Hey @guilhermebruzzi we think for now that's fine, we will do some break change in the next version of amplify. Did you verify this working?

Comment (guilhermebruzzi) [reply packages/aws-amplify-react-native/src/PushNotification/PushNotification.js]: @mlabieniec @powerful23 Yes, it's working for the default app here: https://github.com/aws-samples/aws-mobile-react-native-starter#getting-started-

I installed it without any modifications, made the patch above directly on the node_modules and it kept working, including the new context! :)

**Pull request merged**